src/effects/Reverb_libSoX.h

```diff
@@ -218,6 +268,174 @@ typedef struct {
    float * out[2];
 } reverb_t;
 
+
+// Resizes a filter, resetting its contents
+static void filter_t_resize_resetting(filter_t* p, size_t newSize)
+{
+   memset(p->buffer, 0, newSize * sizeof(float));
+   p->ptr = p->buffer;
+   p->store = 0;
+   p->size = newSize;
+}
+
+
+// Resizes a filter, trying to keep as much of its history as possible.
+static void filter_t_resize_preserving(filter_t* p, size_t newSize)
+{
+   // Imagine we have this filter_t as input:
+   //
+   //                  ptr goes from right to left and when falling off the left side, it warps back to the right;
+   //    <--ptr--      it reads the most distant sample in the past, and after reading it will write a new sample.
+   //        |         Imagine someone just recorded "A B C D E", and now we are going to read "A", but a resize request comes.
+   //        v         
+   //  | C B A E D |
+   //
+   
+   // Depending on the new requested size and where ptr is, we can have these cases:
+   // 
+   //        v
+   //  | C B A 0 E D |  bigger size: right-shift what is right of ptr,
+   //                   and fill the resulting gap with zeros.
+   //
+   //                   (with variant: ptr is already at the old right edge - then shift nothing, just append zeros)
+   //        v
+   //  | C B A D |      smaller size, and ptr is within it (but not at the edge): left-shift what is to the right of ptr,
+   //                   discarding samples.
+   // 
+   //        V
+   //  | C B A |        smaller size, and ptr would be at the new right edge; do nothing.
+   // 
+   //      v
+   //  | B A |          smaller size, and ptr would be beyond the new right edge: left-shift what is ahead of ptr,
+   //                   AND move ptr to size-1
+
+   assert(newSize > 0);
+
+   const int    sizeDiff = (newSize - p->size);
```

Comment:
DANGER -- mixing unsigned and signed arithmetic.

I think `newSize` and `p->size` which are both `size_t` should first be each cast to `ssize_t` before the subtraction,  which may make a negative value.

Reply:
Totally right - done it.